lua/metrics_helper.lua

```diff
@@ -106,6 +106,18 @@ local function emit_cache_metrics(start_time, end_time, namespace, response, sta
             status,
             cache_status
         )
+    elseif
+        string.sub(namespace, 1, 21) == 'yelp-main.internalapi'
+        and ngx.var.request_uri == '/ads/v1/write_to_caches_and_get_ad_opportunity_logging_data'
+    then
+        -- Emit additional internalapi metrics to collect data for DAR-1603
+        emit_request_timing(
```

Comment:
Just a question: is this a non-blocking operation?

Reply:
@giuliano108 it's a blocking operation (as well as in the routing service!) because it uses UDP from the [luasocket](https://w3.impa.br/~diego/software/luasocket/) library.
However, given that UDP is non-blocking itself it does not cause much harm.